torch/_dynamo/variables/builder.py

```diff
@@ -964,13 +971,13 @@ def wrap_tensor(self, value: torch.Tensor):
     def wrap_numpy_ndarray(self, value):
         assert isinstance(value, np.ndarray)
 
-        source = self.get_source()
+        source = NumpyTensorSource(self.get_source())
         tensor_value = torch.as_tensor(value)
-
+        tensor_vt = VariableBuilder(self.tx, source)(tensor_value)
```

Comment:
I guess I don't really understand why you can't just pass this source directly into `wrap_fx_proxy_cls`, with the idea that it's going to create a guard for the tensor_value you pass it, but then disgorge a NumpyNdarrayVariable. Creating a throwaway tensor_vt and smuggling the guards off it smacks me the wrong way.

Reply:
It's not great, but it keeps guard preservation 1:1 with the tensor. I was initially going to manually do this, but then I realized that if tensor guards diverge, this would be kind of a weak contract to enforce. This forces them to be the same.